src/app/gm-tools/feature/edit-event-teams-page.component.ts

```diff
@@ -0,0 +1,610 @@
+
+  readonly originalEvent: Signal<EventModel | null> = input.required(); // route resolve data
+
+  // make the DropListIds enum available in component template
+  readonly DropListIds = DropListIds;
+
+  readonly headerText = computed(
+    () => `Edit ${this.originalEvent()?.name ?? ''} Teams`,
+  );
+
+  readonly submitting = signal(false);
+
+  private readonly newTeamTemplate = computed(() => ({
+    created_at: '',
+    event_id: this.eventId(),
+    name: '',
+    updated_at: '',
+  }));
+
+  readonly databaseEventTeams: Signal<EventTeamModel[] | undefined> = computed(
+    () =>
+      this.eventService
+        .eventTeams()
+        ?.filter((eventTeam) => eventTeam.event_id === this.eventId())
+        ?.sort((a, b) => (a.seed ?? 0) - (b.seed ?? 0)),
+  );
+
+  readonly localSortedEventTeams: WritableSignal<EventTeamModel[] | undefined> =
+    signal(structuredClone(this.databaseEventTeams()));
+
+  readonly databaseEventParticipants: Signal<
+    EventParticipantWithPlayerInfo[] | undefined
+  > = computed(() =>
+    this.eventService
+      .eventParticipantsWithPlayerInfo()
+      ?.filter((eventParticipant) =>
+        this.databaseEventTeams()
+          ?.map((eventTeam) => eventTeam.id)
+          .includes(eventParticipant.team_id),
+      ),
+  );
+
+  readonly localEventParticipants: WritableSignal<
+    EventParticipantWithPlayerInfo[] | undefined
+  > = signal([
+    ...(this.players()
+      ?.filter(
+        (player) =>
+          !this.databaseEventParticipants()
+            ?.map((eventParticipant) => eventParticipant.player_id)
+            .includes(player.player_id),
+      )
+      .map(
+        (player) =>
+          ({
+            participant_id: -player.player_id,
+            team_id: TeamIds.UnassignedTeam,
+            player_id: player.player_id,
+            display_name: player.display_name,
+            avatar_url: player.avatar_url,
+          }) as EventParticipantWithPlayerInfo,
+      ) ?? ([] as EventParticipantWithPlayerInfo[])),
+    ...(structuredClone(this.databaseEventParticipants()) ??
+      ([] as EventParticipantWithPlayerInfo[])),
+  ]);
+
+  readonly eventTeamsWithParticipantInfo = computed(() => [
+    // include a "team" for unassigned players so the container type matches for drag/drop
+    {
+      ...this.newTeamTemplate(),
+      id: TeamIds.UnassignedTeam,
+      participants: this.localEventParticipants()?.filter(
+        (eventParticipant) =>
+          eventParticipant.team_id === TeamIds.UnassignedTeam,
+      ),
+    } as EventTeamWithParticipantInfo,
+
+    ...(this.localSortedEventTeams()?.map((eventTeam) => ({
+      ...eventTeam,
+      participants: this.localEventParticipants()?.filter(
+        (eventParticipant) => eventParticipant.team_id === eventTeam.id,
+      ),
+    })) ?? []),
+
+    // include a "new team" so the container type matches for drag/drop
+    {
+      ...this.newTeamTemplate(),
+      id: TeamIds.NewTeam,
+    } as EventTeamWithParticipantInfo,
```

Comment:
My only issue with this is that `eventTeamsWithParticipantInfo` gets its value from local changes (i.e. `localSortedEventTeams`) and includes two extra "teams": one for unassigned players and one for creating new teams.  That way, t serves as a "view model" for the HTML to make it easier to drag/drop and show event participants with their associated teams.  So we couldn't just create this in the event service as-is because it has some extra stuff specific to the Edit Teams page.

I think the best approach would be to add `eventTeamsWithParticipantInfo` to the event service, change `databaseEventTeams` and `localSortedEventTeams` to get their value from this signal in event service, and continue to use the view model the same way.

Reply:
Gotcha. I didn’t take the time to fully digest what was going on here. Do whatever you think is best 😁

Reply:
I ended up leaving it how it was, it was causing more problems than it was worth to change the `WritableSignal` version of this on the Edit Teams page.